src/SessionManager.php

```diff
@@ -130,9 +130,13 @@ public function start($preserveStorage = false)
 
         session_start();
 
-        if ($oldSessionData instanceof \Traversable
```

Comment:
Looks like this condition is removed. What then in case when `$oldSessionData instanceof \Traversable` (`=== true`) ?

Reply:
May I ask what _should_ be done in such case? Is there a PHP (or ZF) function that converts `Traversable` to `array` or should I write something like:

```php
if ($oldSessionData instanceof \Traversable) {
    $oldSessionDataArray = [];

    foreach ($oldSessionData as $oldSessionDataKey => $oldSessionDataValue) {
        $oldSessionDataArray[$oldSessionDataKey] = $oldSessionDataValue;
    }

    $oldSessionData = $oldSessionDataArray;
}
```

Reply:
See: http://php.net/manual/en/function.iterator-to-array.php

Reply:
Yeah, I’ve updated the patch.

Reply:
One more thing to note: `iterator_to_array` is not recursive

@gscscnd Can you provide an example when `$_SESSION` is not an array?

Reply:
That is, before being passed to `ArrayUtils::merge`? I’ve encountered #10 when implementing “remember me” functionality — I had to retrieve `SessionManager` and I wasn’t able to do it until I’ve provided the following (minimal) configuration:

```php
return [
    'session_config' => [],
    'session_storage' => ['type' => Zend\Session\SessionStorage::class],
];
```

What happened is probably the same as described in #10. Yet another workaround: 8trust/zend-session@c5e523a4b661e3fb336bd009dc21f8edfe687270

Reply:
@gscscnd thanks, now I see the issue, basically it's not possible to use other session storage than (session)array(storage), right?

Reply:
I haven’t tried all storage variants, but it seems `SessionManager` is not fully compatible with `SessionStorage` (and perhaps not only this), so either of them should be somehow fixed :confused: